shared/utils/yaml-to-shorthand.py

```diff
@@ -45,6 +37,14 @@ def add_sub_element(parent, tag, data):
     return element
 
 
+def add_warning_elements(element, warnings):
+    for warning_dict in warnings:
+        warning = add_sub_element(element, "warning", warning_dict.values()[0])
+        warning.set("category", warning_dict.keys()[0])
```

Comment:
I think I get the syntax now but this *needs* to be explained here. the code looks like a total WTF

please also assert that there is exactly one key and exactly one value